pandas/core/reshape/pivot.py

```diff
@@ -424,42 +435,48 @@ def _convert_by(by):
 
 @Substitution("\ndata : DataFrame")
 @Appender(_shared_docs["pivot"], indents=1)
-def pivot(data: "DataFrame", index=None, columns=None, values=None) -> "DataFrame":
+def pivot(
+    data: "DataFrame",
+    index: Optional[Union[Label, Sequence[Label]]] = None,
+    columns: Optional[Union[Label, Sequence[Label]]] = None,
+    values: Optional[Union[Label, Sequence[Label]]] = None,
+) -> "DataFrame":
     if columns is None:
         raise TypeError("pivot() missing 1 required argument: 'columns'")
-    columns = columns if is_list_like(columns) else [columns]
+
+    columns = com.convert_to_list_like(columns)
 
     if values is None:
-        cols: List[str] = []
+        cols: List[Label] = []
         if index is None:
             pass
-        elif is_list_like(index):
-            cols = list(index)
         else:
-            cols = [index]
+            cols = com.convert_to_list_like(index)
```

Comment:
i think this block can now be further simplified to avoid the pass statement, also probably don't need the variable annotation?

Reply:
yeah, changed to 
```python
if index is not None:
    cols = com.convert_to_list_like(index)
```

Reply:
still need

```
else:
    cols = []
```
reversing the order of assignment so that the empty list is further down (in static analysis terms) should negate the need for the variable annotation since the empy list is compatible with the list defined above

Reply:
yeah, just found out it the failed test